framework/src/neml2/userobjects/NEML2FEInterpolation.C

```diff
@@ -0,0 +1,415 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef NEML2_ENABLED
+
+// libmesh includes
+#include "libmesh/petsc_vector.h"
+
+// Torch includes
+#include <ATen/ops/from_blob.h>
+
+// NEML2 includes
+#include "neml2/tensors/functions/discretization/scatter.h"
+#include "neml2/tensors/functions/discretization/interpolate.h"
+
+// MOOSE includes
+#include "NEML2FEInterpolation.h"
+
+using namespace libMesh;
+
+registerMooseObject("MooseApp", NEML2FEInterpolation);
+
+InputParameters
+NEML2FEInterpolation::validParams()
+{
+  InputParameters params = ElementUserObject::validParams();
```

Comment:
this is an element UO so it can be block restricted
I think you should check the variables at hand share that block restriction

Reply:
Why wouldn't that be done in a base class, such as `ElementUserObject` or `Coupleable`? Doesn't seem right to delegate that responsibility to a leaf class.

Reply:
if you were using COupleable it would be done that way.
But this isnt I think, it's building and keeping track of its own list of variables